front/collect_file.form.php

```diff
@@ -48,6 +48,13 @@
            && $_POST['filter_name'] != ''
     ) {
         $_POST['filter_' . $_POST['filter_nametype']] = $_POST['filter_name'];
+
+        //set null if needed
+        $_POST['filter_nametype'] == 'iname' ? $_POST['filter_name'] = null : $_POST['filter_iname'] = null;
```

Comment:
Ternary notation is usally used to affect values to variable that is declared in the beginning of the line. Using it this way is not common in GLPI codebase and could result in misnderstanding of the code (was my case on first reading).

```suggestion
        if ($_POST['filter_nametype'] == 'iname') {
            $_POST['filter_name'] = null;
        } else {
            $_POST['filter_iname'] = null;
        }
```